### ISSUE TITLE
Server unerreichbar - Meine VA sehr langsam

### ISSUE THREAD

**Issue by Davka** [elan-ev/studip-opencast-plugin#551]
Wenn der OC-Server (bspw. weil Elasticsearch Probleme macht) nicht erreichbar ist, lädt die Seite Meine VA sehr langsam bis gar nicht.

**Comment by Rillke** [comment]
Ja, und wenn man nicht genug PHP-Prozesse hinter einem Load-Balancer hat, werden alle Benutzer irgendwann nach Bad Gateway umgeleitet.

**Comment by tgloeggl** [comment]
Was ist denn ein sinnvoller Timeout-Wert? Ich habe jetzt 3 Sekunden gewählt. Falls das Probleme macht, muss der Wert nochmal angepasst oder konfigurierbar werden.

**Comment by ferishili** [comment]
To address the issue effectively, the implementation plan includes the following steps:

* **Introduce configurable timeouts:** Ensure that timeout values, particularly `connection_timeout`, are configurable and set to reasonable defaults. A 60-second timeout is too long for most scenarios and should be adjusted accordingly, why not configurable?

* **Add retry middleware:** Implement retry logic to handle transient failures and improve resilience, especially in cases where Opencast becomes temporarily unresponsive.

* **Enhance error handling:** Review and improve error handling in critical areas to ensure graceful degradation and clearer diagnostics during failures.
